### PR TITLE
[TSCUtility] Add support for creating a persistent cache

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by neonichu** [comment]
why do we even need a cache

**Comment by aciidb0mb3r** [comment]
@swift-ci test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 xcuserdata/
 .swiftpm
 build
+.vscode
```
Comment (neonichu) [inline .gitignore:8]: 🤔

Comment (aciidb0mb3r) [reply .gitignore:8]: best code

**Pull request merged**